packages/platform-server/src/agents/threads.controller.ts

```
@@ -447,6 +450,61 @@ export class AgentsThreadsController {
     return { items };
   }
 
+  @Delete('threads/:threadId/queued-messages')
+  async clearQueuedMessages(@Param('threadId') threadId: string) {
+    const thread = await this.persistence.getThreadById(threadId);
+    if (!thread) throw new NotFoundException({ error: 'thread_not_found' });
+
+    const assignedAgentNodeId = typeof thread.assignedAgentNodeId === 'string' ? thread.assignedAgentNodeId.trim() : '';
+    if (!assignedAgentNodeId) {
+      return { clearedCount: 0 } as const;
+    }
+
+    const liveNodes = this.runtime.getNodes();
+    const agentNodes = liveNodes.filter((node) => isAgentLiveNode(node, this.templateRegistry));
+    if (agentNodes.length === 0) {
+      return { clearedCount: 0 } as const;
+    }
+
+    const liveAgentNode = agentNodes.find((node) => node.id === assignedAgentNodeId);
+    if (!liveAgentNode) {
+      return { clearedCount: 0 } as const;
+    }
+
+    const instance = liveAgentNode.instance;
+    if (!isAgentRuntimeInstance(instance) || !hasQueueManagementCapability(instance)) {
+      return { clearedCount: 0 } as const;
+    }
+
+    try {
+      const cleared = instance.clearQueuedMessages(threadId);
+      return { clearedCount: Number.isFinite(cleared) ? Math.max(0, Math.trunc(cleared)) : 0 } as const;
+    } catch (error) {
+      const stack = error instanceof Error ? error.stack : undefined;
+      this.logger.warn(
+        `clearQueuedMessages runtime failure thread=${threadId} agent=${assignedAgentNodeId}`,
+        stack,
+        AgentsThreadsController.name,
+      );
+      return { clearedCount: 0 } as const;
```

[major] If  throws we swallow it and respond . From the client’s perspective that looks like a successful cancel, so we optimistically remove the entry and the user never hears about the failure. For real runtime errors we should surface a 500 (e.g. throw ) so the UI can restore the queued message and show the error toast. Returning zero silently hides the problem and violates the error-handling requirement in #1252.